File: CS/myTreeNode.cs
using System.Drawing;
using System.Data;
using System.Windows.Forms;
using System.Collections;
using Microsoft.VisualBasic;
using System.Diagnostics;
using System;

namespace bftm
{
	public class myTreeNode : TreeNode
	{
		
		
		int m_ID;
		bool m_Active;
		
		public myTreeNode(string Name) : base(Name)
		{
		}
		
		public myTreeNode(string Name, string Type, int ID, bool Active) : base(Name)
		{
			base.Tag = Type;
			m_ID = ID;
			m_Active = Active;
		}
		
		public myTreeNode(string Name, string Type, int ID, bool Active, int ImageIndex, int SelectedImageIndex) : base(Name, ImageIndex, SelectedImageIndex)
		{
			base.Tag = Type;
			m_ID = ID;
			m_Active = Active;
		}
		
		public int ID
		{
			get
			{
				return m_ID;
			}
			set
			{
				m_ID = value;
			}
		}
		
		public bool isActive
		{
			get
			{
				return m_Active;
			}
			set
			{
				m_Active = value;
			}
		}
		
	}
	
	
}

File: myTreeNode.vb
Public Class myTreeNode
    Inherits TreeNode

    Dim m_ID As Integer
    Dim m_Active As Boolean

    Sub New(ByVal Name As String)
        MyBase.New(Name)
    End Sub

    Sub New(ByVal Name As String, ByVal Type As String, ByVal ID As Integer, ByVal Active As Boolean)
        MyBase.New(Name)
        MyBase.Tag = Type
        m_ID = ID
        m_Active = Active
    End Sub

    Sub New(ByVal Name As String, ByVal Type As String, ByVal ID As Integer, ByVal Active As Boolean, _
        ByVal ImageIndex As Integer, ByVal SelectedImageIndex As Integer)
        MyBase.New(Name, ImageIndex, SelectedImageIndex)
        MyBase.Tag = Type
        m_ID = ID
        m_Active = Active
    End Sub

    Property ID() As Integer
        Get
            Return m_ID
        End Get
        Set(ByVal Value As Integer)
            m_ID = Value
        End Set
    End Property

    Property isActive() As Boolean
        Get
            Return m_Active
        End Get
        Set(ByVal Value As Boolean)
            m_Active = Value
        End Set
    End Property

End Class

